javascript/javascript2/week2/02-woking-with-movies/script.js

```diff
@@ -0,0 +1,106 @@
+//Q1) short title--------------------------------
+const shortTitledMovies = movies.filter((movie) => movie.title.length < 5);
+console.log(shortTitledMovies.length);
+
+// Q2) long movie titles---------------------------
+const longMovieTitles = movies.filter((movie) => movie.title.length > 5);
+// console.log(`movies with long titles ${JSON.stringify(longMovieTitles)}`);
+
+// Q3)made between 1980-1989-------------------------------
+
+let number = 0;
+const moviesMade = movies.filter(
+  (movie) => movie.year >= 1980 && movie.year <= 1989
+);
+// console.log(`made between 1980-1989 ${JSON.stringify(moviesMade)}`);
+
+//Q4) extra key called tag---------------------------------------
+const tag = movies.map(function (movie) {
+  if (movie.rating >= 7) {
+    movie.tag = "Good";
+  } else if (movie.rating < 7 && movie.rating >= 4) {
+    movie.tag = "Average";
+  } else if (movie.rating < 4) {
+    movie.tag = "Bad";
+  }
+  return movie;
```

Comment:
```suggestion
  let tag = null;
  
  if (movie.rating >= 7) {
    tag = "Good";
  } else if (movie.rating < 7 && movie.rating >= 4) {
    tag = "Average";
  } else if (movie.rating < 4) {
    tag = "Bad";
  }
  
  return {
    ...movie,
    tag,
  };
```

It's a good practice to avoid modifying arguments to a function!